march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_subgait.py

```diff
@@ -27,38 +46,38 @@ def empty_subgait(cls, gait_generator, robot, duration=8, gait_type='walk_like',
                 rospy.logwarn('Skipping joint ' + urdf_joint.name + ' because it has no limits.')
                 continue
 
-            default_setpoints = [
-                ModifiableSetpoint(0, 0, 0),
-                ModifiableSetpoint(duration, 0, 0),
-            ]
+            default_setpoints = [ModifiableSetpoint(0, 0, 0), ModifiableSetpoint(duration, 0, 0)]
+
             limits = Limits(urdf_joint.safety_controller.soft_lower_limit,
                             urdf_joint.safety_controller.soft_upper_limit,
                             urdf_joint.limit.velocity,
                             urdf_joint.limit.effort,
                             urdf_joint.safety_controller.k_position,
                             urdf_joint.safety_controller.k_velocity)
-            joint = ModifiableJointTrajectory(urdf_joint.name,
-                                              limits,
-                                              default_setpoints,
-                                              duration,
-                                              gait_generator,
-                                              )
+
+            joint = ModifiableJointTrajectory(urdf_joint.name, limits, default_setpoints, duration, gait_generator)
+
             joint_list.append(joint)
-        return cls(joint_list, duration, gait_type, gait_name, subgait_name, version, description)
+
+        return cls(joint_list, duration)
```

Comment:
in my opinion it would be way more logical to pass these standard values as defaults to the constructor since this structure would be identical to the subgait in the march_shared classes.

Reply:
But you do not need to overload the constructor if you add the default values here right? Besides, these defaults are used for the empty_subgait method only, so it would make sense to just define them there I think